network/bridge_networkclient_linux.go

```diff
@@ -50,18 +50,31 @@ func (client *LinuxBridgeClient) CreateBridge() error {
 
 func (client *LinuxBridgeClient) AddRoutes(nwInfo *NetworkInfo, interfaceName string) error {
 	if client.nwInfo.IPAMType == AzureCNS {
+
+		// Add snat Rules
+		gwIP := client.nwInfo.Options[SNATIPKey]
+		if gwIP == nil {
+			return fmt.Errorf("Host gateway IP in Options not set")
+		}
+
+		gatewayIP := net.ParseIP(gwIP.(string))
+		if gatewayIP == nil {
+			return fmt.Errorf("Invalid host gateway IP: %+v", gwIP)
+		}
 		// add pod subnet to host
 		devIf, _ := net.InterfaceByName(interfaceName)
 		ifIndex := devIf.Index
-		family := netlink.GetIpAddressFamily(Ipv4DefaultRouteDstPrefix.IP)
+		family := netlink.GetIpAddressFamily(gatewayIP)
 
 		nlRoute := &netlink.Route{
 			Family:    family,
 			Dst:       &client.nwInfo.PodSubnet.Prefix,
-			Gw:        Ipv4DefaultRouteDstPrefix.IP,
```

Comment:
Seems like instead of adding a rule we're replacing a rule. Also what is the podsubnet here is it actually the node subnet (what I would think of in aks-rp as vnetsubnetid given to aks?)